tools/IceRpc.BuildTelemetry.Reporter/Program.cs

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) ZeroC, Inc.
+
+using IceRpc;
+using IceRpc.BuildTelemetry.Reporter;
+using System.Net.Security;
+using System.Security.Cryptography.X509Certificates;
+
+const int timeout = 3000; // The timeout for the complete telemetry upload process.
+const string uri = "icerpc://localhost"; // The URI of the server.
+
+// Load the root CA certificate
+using var rootCA = new X509Certificate2("certs/cacert.der");
+var clientAuthenticationOptions = new SslClientAuthenticationOptions
+{
+    RemoteCertificateValidationCallback = (sender, certificate, chain, errors) =>
+    {
+        using var customChain = new X509Chain();
+        customChain.ChainPolicy.RevocationMode = X509RevocationMode.NoCheck;
+        customChain.ChainPolicy.DisableCertificateDownloads = true;
+        customChain.ChainPolicy.TrustMode = X509ChainTrustMode.CustomRootTrust;
+        customChain.ChainPolicy.CustomTrustStore.Add(rootCA);
+        return customChain.Build((X509Certificate2)certificate!);
+    }
+};
+
+try
+{
+    // Create a cancellation token source to cancel the telemetry upload if it
+    // takes too long.
+    using var cts = new CancellationTokenSource(TimeSpan.FromMilliseconds(timeout));
```

Comment:
Why not use the deadline interceptor?

Reply:
We want to use this token for the invocation + the shutdown.